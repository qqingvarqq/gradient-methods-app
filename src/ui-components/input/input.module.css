.root {
  border: none;
  font-size: 24px;
  background: none;
  min-width: 100px;
  height: 28px;
  padding: 0;
  transition: box-shadow 0.3s ease;
  margin: 0;
  box-sizing: border-box;
}
.root:hover {
  box-shadow: 0px 0px 2px 1px rgba(0, 0, 0, 0.3);
}
.root:focus {
  box-shadow: 0px 0px 2px 1px rgba(0, 0, 0, 0.3);
  outline: none;
}

input.root::-webkit-outer-spin-button,
input.root::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

input.root[type='number'] {
  -moz-appearance: textfield; /* Firefox */
}
.error,
.error:hover,
.error:focus {
  box-shadow: 0px 0px 2px 1px rgba(253, 76, 76, 0.7);
}
