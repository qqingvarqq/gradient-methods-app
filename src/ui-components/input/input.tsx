import * as React from 'react';
import styles from './input.module.css';

export interface InputProps
  extends React.DetailedHTMLProps<
    React.InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  > {
  error?: boolean;
}
export const Input: React.FC<InputProps> = props => {
  const {className = '', error = false, ...rest} = props;
  const errorClassName = error ? styles.error : '';
  return (
    <input
      className={`${styles.root} ${errorClassName}  ${className} `}
      {...rest}></input>
  );
};
