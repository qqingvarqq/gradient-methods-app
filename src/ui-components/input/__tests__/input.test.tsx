import * as React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {Input} from '../input';

describe('Input', () => {
  it('should render without crashing', () => {
    const {asFragment} = render(<Input type="text" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render without crashing with type="number"', () => {
    const {asFragment} = render(<Input type="number" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render without crashing with error prop', () => {
    const {asFragment} = render(<Input type="number" error />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should fire onChange event callback', () => {
    const onChange = jest.fn();
    const {getByPlaceholderText} = render(
      <Input type="text" value="" placeholder="testInput" onChange={onChange} />
    );
    const input = getByPlaceholderText('testInput');
    fireEvent.change(input, {target: {value: 'test'}});
    expect(onChange).toBeCalled();
  });
});
