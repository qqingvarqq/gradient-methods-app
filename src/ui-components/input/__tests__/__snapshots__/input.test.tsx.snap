// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render without crashing 1`] = `
<DocumentFragment>
  <input
    class="root    "
    type="text"
  />
</DocumentFragment>
`;

exports[`Input should render without crashing with error prop 1`] = `
<DocumentFragment>
  <input
    class="root error   "
    type="number"
  />
</DocumentFragment>
`;

exports[`Input should render without crashing with type="number" 1`] = `
<DocumentFragment>
  <input
    class="root    "
    type="number"
  />
</DocumentFragment>
`;
